pkg/server/wire.go

```diff
@@ -379,6 +379,20 @@ var wireSet = wire.NewSet(
 	wire.Bind(new(oauthtoken.OAuthTokenService), new(*oauthtoken.Service)),
 )
 
+var wireCLISet = wire.NewSet(
```

Comment:
this might be a good time to add some comments about the difference between these sets, and suggestions for what new things go where (if you have an idea. I don't).

Reply:
Ideally, everything should go into `wireBasicSet`. CLISet I hope would not be needed soon (need to look into enterprise) and it could be instantiated from the same BasicSet graph as would the Server.
Now, wireTestSet is a question. I'd prefer to see it gone and let unit tests inject dependencies (mock) manually, as most of them do already. Integration tests on the other hand should rather use real implementations when possible. We should be able to move prefs and sqlstore into the basic set. NGAlert and Notifications are terra incognita to me, but I hope there would be a solution to mock them in a similar manner without a separate wire graph.
TLDR: I hope there would be one wireSet once this is all finished. What we have now is there "for legacy reasons".

Reply:
I'm still not sure the extent to which we interact with these sets on a day-to-day basis, but I second documenting these guidelines somehow. Your comment seems like a good overview already. The documentation could be something along the lines of the [Queries with Result Fields](https://github.com/grafana/grafana/blob/15804234d6ea8d803def4edf802d142e418979f1/contribute/backend/communication.md#queries-with-result-fields) section, which describes how to refactor away Result fields. I suppose the guidelines you describe could be added to the [Services](https://github.com/grafana/grafana/blob/15804234d6ea8d803def4edf802d142e418979f1/contribute/backend/services.md) section, which already includes an introduction to wire.